src/bit-systems/mixer-animatable.ts

```diff
@@ -0,0 +1,33 @@
+import { addComponent, defineQuery, enterQuery, exitQuery, removeComponent } from "bitecs";
+import { AnimationMixer } from "three";
+import { MixerAnimatable, MixerAnimatableInitialize, MixerAnimatableData, Object3DTag } from "../bit-components";
+import { HubsWorld } from "../app";
+
+const initializeQuery = defineQuery([MixerAnimatableInitialize, Object3DTag]);
+const initializeEnterQuery = enterQuery(initializeQuery);
+const mixerQuery = defineQuery([MixerAnimatable, Object3DTag]);
+const mixerExitQuery = exitQuery(mixerQuery);
+
+export function mixerAnimatableSystem(world: HubsWorld): void {
+  initializeEnterQuery(world).forEach(eid => {
```

Comment:
Confused why we need 2 components here (MixerAnimatableInitialize and MixerAnimatable). Can't we just create the Mixer on enter query for MixerAnimatable? That's what I did in the quick stub of this I am using in my behavior graph branch and it seems to work well https://github.com/mozilla/hubs/blob/behavior-graphs-spike/src/bit-systems/behavior-graph.ts#L82-L101. Since we have strong control over exectuion order we can guarantee that the mixer is set up before any other code using it runs (both in this system and the loop-animation system).

I would also prefer to keep the map on the MixerAnimatable component itself instead of having it floating seperatly. It makes it feel more like any other component property: https://github.com/mozilla/hubs/blob/behavior-graphs-spike/src/bit-components.js#L185-L190. With the type comment I have there it also typechecks just fine too.